src/Data/String/CodePoints.js

```diff
@@ -0,0 +1,119 @@
+"use strict";
+/* global Symbol */
+
+var hasArrayFrom = typeof Array.from === "function";
+var hasStringIterator =
+  typeof Symbol !== "undefined" &&
+  Symbol != null &&
+  typeof Symbol.iterator !== "undefined" &&
+  typeof String.prototype[Symbol.iterator] === "function";
+var hasFromCodePoint = typeof String.prototype.fromCodePoint === "function";
+var hasCodePointAt = typeof String.prototype.codePointAt === "function";
+
+exports._unsafeCodePointAt0 = function (fallback) {
+  return hasCodePointAt
+    ? function (str) { return str.codePointAt(0); }
+    : fallback;
+};
+
+exports._codePointAt = function (fallback) {
+  return function (Just) {
+    return function (Nothing) {
+      return function (unsafeCodePointAt0) {
+        return function (index) {
+          return function (str) {
+            var length = str.length;
+            if (index < 0 || index >= length) return Nothing;
+            if (hasArrayFrom) {
```

Comment:
Sort of carrying on from our discussion before; why do we expect `Array.from` to be faster than String iteration in general? I think people will expect it to be fast to call `codePointAt` with a small argument even on very large strings, and I'm not aware of reasons to prefer `Array.from`; can you expand a little more on this please?